plugins/rebalance/rest.go

```diff
@@ -0,0 +1,276 @@
+	//TODO
+
+	//A simple transaction to start rebalance
+	txn := transaction.NewTxn(ctx)
+	defer txn.Cleanup()
+	lock, unlock, err := transaction.CreateLockSteps(volname)
+	if err != nil {
+		restutils.SendHTTPError(r.Context(), w, http.StatusInternalServerError, err.Error(), api.ErrCodeDefault)
+		return
+	}
+
+	txn.Nodes = vol.Nodes()
+	txn.Steps = []*transaction.Step{
+		lock,
+		{
+			DoFunc: "rebal-start.Commit",
+			Nodes:  txn.Nodes,
+		},
+		{
+			DoFunc: "rebal-start.StoreVolume",
+			Nodes:  []uuid.UUID{gdctx.MyUUID},
+		},
+		unlock,
+	}
+
+	err = txn.Ctx.Set("volname", volname)
+	if err != nil {
+		logger.WithError(err).Error("failed to set request in transaction context")
+		restutils.SendHTTPError(ctx, w, http.StatusInternalServerError, err.Error(), api.ErrCodeDefault)
+		return
+	}
+
+	rebalance.Volname = volname
+	rebalance.Status = rebalanceapi.GfDefragStatusStarted
+	rebalance.Cmd = rebalanceapi.GfDefragCmdStart
+	setcommithash(rebalance)
+	rebal, err := createRebalanceInfo(rebalance)
+	if err != nil {
+		logger.WithError(err).Error("failed to create Rebalance info")
+		restutils.SendHTTPError(ctx, w, http.StatusInternalServerError, err.Error(), api.ErrCodeDefault)
+		return
+	}
+
+	err = txn.Ctx.Set("rinfo", rebal)
+	if err != nil {
+		logger.WithError(err).Error("failed to set rebalance info in transaction context")
+		restutils.SendHTTPError(ctx, w, http.StatusInternalServerError, err.Error(), api.ErrCodeDefault)
+		return
+	}
+
+	err = txn.Do()
+	if err != nil {
+		logger.WithFields(log.Fields{
+			"error":   err.Error(),
+			"volname": volname,
+		}).Error("failed to start rebalance on volume")
+		restutils.SendHTTPError(r.Context(), w, http.StatusInternalServerError, err.Error(), api.ErrCodeDefault)
+		return
+	}
+
+	if err = txn.Ctx.Get("rinfo", &rebal); err != nil {
+		restutils.SendHTTPError(ctx, w, http.StatusInternalServerError, "failed to get rebalanceinfo", api.ErrCodeDefault)
+		return
+	}
+
+	txn.Ctx.Logger().WithField("volname", rebal.Volname).Info("rebalance started")
+	restutils.SendHTTPResponse(ctx, w, http.StatusOK, rebal)
+}
+
+func rebalanceStop(w http.ResponseWriter, r *http.Request) {
+	ctx := r.Context()
+	logger := gdctx.GetReqLogger(ctx)
+	rebalance := new(rebalanceapi.RebalanceInfo)
+
+	// collect inputs from url
+	volname := mux.Vars(r)["volname"]
+
+	// Validate rebalance command
+	vol, err := volume.GetVolume(volname)
+	if err != nil {
+		restutils.SendHTTPError(r.Context(), w, http.StatusNotFound, "Invalid volume", api.ErrCodeDefault)
+		return
+	}
+
+	rebalinfo, err := GetRebalanceInfo(volname)
+	if err != nil {
+		restutils.SendHTTPError(ctx, w, http.StatusNotFound, err.Error(), api.ErrCodeDefault)
+		return
+	}
+
+	// Check rebalance process is started or not
+	if rebalinfo.Status != rebalanceapi.GfDefragStatusStarted {
+		restutils.SendHTTPError(r.Context(), w, http.StatusBadRequest, "Rebalance process is not started",
+			api.ErrCodeDefault)
+		return
+	}
+
+	if vol.DistCount == 1 {
```

Comment:
This check is not required. Rebalance already started implies that it is already satisfying this requirement.